### PR TITLE
[FIX] account: fix payment term installments ordering

### PR DESCRIPTION
**Pull request by casp-odoo**
The payment terms printed on the invoice are ordered by date.

However, the order was applied on the string version of the date, which caused issues.

Fixed this by formatting the date to string after the ordering.

### CODE REVIEW

**Review by william-andre — APPROVED** [review]
Another solution would have been
```diff
                payment_term_details = []
-               for line in invoice.line_ids.filtered(lambda l: l.display_type == 'payment_term'):
+               for line in invoice.line_ids.filtered(lambda l: l.display_type == 'payment_term').sorted('date'):
                    payment_term_details.append({
                        'date': line.date_maturity,
                        'amount': sign * line.amount_currency,
                        'discount_date': format_date(self.env, line.discount_date),
                        'discount_amount_currency': sign * line.discount_amount_currency,
                    })
-               invoice.payment_term_details = sorted(payment_term_details, key=lambda t: t.get('date'))
+               invoice.payment_term_details = payment_term_details
```

@robodoo r+

**Comment by william-andre** [comment]
@robodoo r-
Actually I'd lie that solution :)

**Comment by casp-odoo** [comment]
@william-andre Alright, I did that :)

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -1235,14 +1235,16 @@ def _compute_payment_term_details(self):
...
-                        'date': format_date(self.env, line.date_maturity),
+                        'date': line.date_maturity,
                         'amount': sign * line.amount_currency,
                         'discount_date': format_date(self.env, line.discount_date),
                         'discount_amount_currency': sign * line.discount_amount_currency,
                     })
-                invoice.payment_term_details = sorted(payment_term_details, key=lambda t: t.get('date'))
+                invoice.payment_term_details = payment_term_details
+                for pay_term in invoice.payment_term_details:
+                    pay_term['date'] = format_date(self.env, pay_term.get('date'))
```
Comment (william-andre) [inline addons/account/models/account_move.py:1247]: But you don't need this diff now, that's the point :sweat_smile: 
You can format directly in the first loop

Comment (casp-odoo) [reply addons/account/models/account_move.py:1247]: Yeah, went a bit too fast on this one :sweat_smile: 
I'll change that in a bit

**Comment by casp-odoo** [comment]
@william-andre does it look better now ? :)

**Review by william-andre — APPROVED** [review]
Perfect!

@robodoo r+

**Pull request closed without merging**